src/components/app/data/hooks/useIsCourseRunUpgradable.js

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { queryCanUpgradeWithLearnerCredit } from '../queries';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+
+/**
+ * Determines whether the given content identifier is contained within the enterprise customer's catalogs.
+ * @returns {Types.UseQueryResult}
+ */
+export default function useIsCourseRunUpgradable(contentIdentifiers, queryOptions = {}) {
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+  const {
+    select, refetchOnMount, refetchOnWindowFocus, ...queryOptionsRest
+  } = queryOptions;
+  return useQuery({
+    ...queryCanUpgradeWithLearnerCredit(enterpriseCustomer.uuid, contentIdentifiers),
+    ...queryOptionsRest,
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
```

Comment:
[curious] Did you run into issues with refetching behavior, or is this more proactively turning it off just in case?

Reply:
This more of a proactive move. I know we briefly discussed turning off some refetching behavior for these calls to avoid multiple API calls `canRedeem` during a user session due to the performance overhead.

Reply:
I think we can keep the re-fetching behavior on so long as it's not causing any problems due to relation with the upgrade-related context provider.

IFF it does cause issues, then we can disable the re-fetching. The re-fetching is a nice-to-have, though, so it does seemingly work without any concerns, great.